### PR TITLE
ray-train: add multi-node EKS integration test (BERT FSDP via KubeRay)

### PR DESCRIPTION
**Pull request by Eren-Jeager123**
## Summary
- Adds a multi-node EKS integration test for the Ray Train DLC
- FSDP fine-tune on GLUE/CoLA via Ray Train + Lightning (8 GPU across 2x g6.12xlarge + CPU head)
- KubeRay RayCluster on the shared `dlc-shared-cluster`, Karpenter auto-provisions nodes
- Data pre-staged on FSx (`/fsx/hf_cache`), runs fully offline
- Validates: loss decreases, 8-rank NCCL formed (multi-node), EFA/Libfabric engaged
- Wired into PR and autorelease workflows (`run-eks-test` and `run-efa-test` both `default: true`)

## Files
- `test/ray-train/eks/raycluster.yml` — KubeRay manifest (CPU head + 2 GPU workers, FSx, EFA)
- `test/ray-train/eks/run_eks_test.sh` — orchestrator (apply, wait, submit, assert, cleanup)
- `test/ray-train/eks/scripts/fsdp_ray.py` — FSDP training script
- `.github/workflows/ray-train.tests-eks.yml` — reusable workflow
- `.github/workflows/ray-train.pipeline.yml` — added `run-eks-test` input + job
- `.github/workflows/ray-train.pr-gpu.yml` — path filter for `test/ray-train/eks/**`

## Infra dependencies (already deployed)
- Karpenter workload pools: `ray-train-head` (m5/m6a) + `ray-train-gpu` (g6.12xlarge, EFA)
- FSx PVC in `ray-train` namespace
- CodeBuild RBAC (`ray-train-ci` group → `ray-train-test-role`)

## Test plan
- [x] RayCluster deploys via Karpenter (nodes auto-provisioned)
- [x] FSDP training completes (loss decreased)
- [x] 8-rank NCCL confirmed (multi-node)
- [x] EFA/Libfabric engaged
- [x] Cleanup: Karpenter deprovisions nodes after test (verified: nodes gone)
- [ ] Security scan (CVE allowlist out of scope — separate PR)

### CODE REVIEW

**Comment by jinyan-li1** [comment]
Could you please link the passing test commit in the PR desc or in this comment?

File: .github/workflows/ray-train.autorelease-gpu.yml [hunk 1]
```diff
@@ -34,6 +34,5 @@ jobs:
     with:
       config-file: ${{ matrix.config_file }}
       tag-suffix: ${{ github.run_id }}
-      run-efa-test: true
```
Comment (jinyan-li1) [inline .github/workflows/ray-train.autorelease-gpu.yml:37]: is this intentionally disabled?

File: test/ray-train/eks/run_eks_test.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#   AWS_REGION      - AWS region (default: us-west-2)
+#   NAMESPACE       - K8s namespace (default: ray-train)
+set -euo pipefail
+
+: "${IMAGE_URI:?IMAGE_URI is required}"
+: "${RAY_VERSION:?RAY_VERSION is required}"
+EKS_CLUSTER="${EKS_CLUSTER:-dlc-shared-cluster}"
+AWS_REGION="${AWS_REGION:-us-west-2}"
+NAMESPACE="${NAMESPACE:-ray-train}"
+CLUSTER_NAME="ray-train-test"
```
Comment (jinyan-li1) [inline test/ray-train/eks/run_eks_test.sh]: nit: would be better to extract from the manifest instead of hardcode again - https://github.com/aws/deep-learning-containers/pull/6438/changes#diff-1e3e01048060be500f7c210ce85e71ad250b3d96199ef79cf206e77e49b0c10aR4

File: .github/workflows/ray-train.autorelease-gpu.yml (hunk 1 above)
Comment (Eren-Jeager123) [reply .github/workflows/ray-train.autorelease-gpu.yml:37]: I changed to `default: true` in `ray-train.pipeline.yml`, so it's not disabled here.

**Review by jinyan-li1 — APPROVED** [review]
